tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -0,0 +1,283 @@
+#include <c10/util/ArrayRef.h>
+#include <fusion.h>
+#include <ir/utils.h>
+#include <ops/all_ops.h>
+#include <tests/cpp/multidevice.h>
+
+namespace nvfuser {
+
+int parseEnvVariable(const char* env_name) {
+  const std::string prefix = "NVFUSER_OVERLAP_";
+  auto prefixed_name = prefix + env_name;
+  auto env = std::getenv(prefixed_name.c_str());
+  if (!env) {
+    return -1;
+  }
+  return std::atoi(env);
+}
+
+bool isEnvVariableDefined(const char* env_name) {
+  return parseEnvVariable(env_name) != -1;
+}
+
+struct OverlapTestParams {
+  // Tensors sizes
+  int64_t M = std::pow(2, 6);
+  int64_t K = std::pow(2, 5);
+  int64_t N = std::pow(2, 4);
+  int64_t S = std::pow(2, 3);
+
+  // network backend type
+  CommunicatorBackend backend_type = CommunicatorBackend::nccl;
+
+  // overlap optimization parameters
+  bool use_different_streams =
+      false; // whether to change CUDA stream at each iteration
+
+  // debug
+  bool debug_print = false;
+
+  void parseEnv() {
+    if (isEnvVariableDefined("LOG2_M")) {
+      M = std::pow(2, parseEnvVariable("LOG2_M"));
+    }
+    if (isEnvVariableDefined("LOG2_K")) {
+      K = std::pow(2, parseEnvVariable("LOG2_K"));
+    }
+    if (isEnvVariableDefined("LOG2_N")) {
+      N = std::pow(2, parseEnvVariable("LOG2_N"));
+    }
+    if (isEnvVariableDefined("LOG2_S")) {
+      S = std::pow(2, parseEnvVariable("LOG2_S"));
+    }
+    if (isEnvVariableDefined("USE_UCC")) {
+      backend_type = CommunicatorBackend::ucc;
+    }
+    if (isEnvVariableDefined("USE_STREAMS")) {
+      use_different_streams = true;
+    }
+    if (isEnvVariableDefined("DEBUG_PRINT")) {
+      debug_print = true;
+    }
+  }
+};
+
+std::ostream& operator<<(std::ostream& out, const OverlapTestParams& params) {
+  std::string indent = "  ";
+  out << "params:{\n"
+      << indent << "backend_type=" << params.backend_type << "\n"
+      << indent << "M=" << params.M << "\n"
+      << indent << "K=" << params.K << "\n"
+      << indent << "N=" << params.N << "\n"
+      << indent << "S=" << params.S << "\n"
+      << indent << "use_different_streams=" << params.use_different_streams
+      << "\n"
+      << "}";
+  return out;
+}
+
+class OverlapTest : public MultiDeviceTest {
+ protected:
+  OverlapTestParams params;
+
+  int64_t num_devices;
+  int64_t my_device_index;
+  at::Tensor ta, tb, tc_unreduced, tc_locally_reduced, tc, tc_expected;
+  // stores the backend
+  c10d::Backend* world_communicator;
+
+  void SetUp() {
+    MultiDeviceTest::SetUp();
+
+    params.parseEnv();
+    num_devices = communicator->size();
+    my_device_index = communicator->deviceId();
+    ASSERT_EQ(params.M % (params.S * num_devices), 0);
+    ASSERT_EQ(params.K % num_devices, 0);
+
+    // Setup the world communicators
+    std::vector<int64_t> devices(num_devices);
+    std::iota(devices.begin(), devices.end(), 0);
+    world_communicator = communicator->getBackendForTeam(
+        devices, /*backend=*/params.backend_type);
```

Comment:
```suggestion
        devices, params.backend_type);
```

The name `backend_type` is pretty clear about which argument of getBackendForTeam it corresponds to.